lc-0278-first-bad-version/src/main/java/solution/Solution.java

```diff
@@ -0,0 +1,37 @@
+package solution;
+
+public class Solution extends VersionControl {
+
+    Solution(int badVersion) {
+        super(badVersion);
+    }
+
+    // Perform binary search:
+    public int firstBadVersion(int n) {
+        int low = 1;
+        int high = n;
+
+        long mid = (low + high) / 2; // declare as long to handle math but 'mid' will always be within 'int'
```

Comment:
I believe this has a bug. You'll first overlow int, and then store the result as long.

But even if done right:
```
long mid = ((long)low + high) / 2
```
it works, but it's messy, and doesn't scale. What if your inputs were long?

You've probably seen this:
```
int mid = low + (high - low) / 2;
```
it's much cleaner. Use it instead.

Reply:
Are you going to fix it or do you just want to merge this?

I don't see a fix for this.